securedrop_client/logic.py

```diff
@@ -177,15 +204,43 @@ def on_authenticate(self, result):
             error = _('There was a problem logging in. Please try again.')
             self.gui.show_login_error(error=error)
 
+    def successful_api_call(self, r, user_callback):
+        logger.info("Hooray, successful API call. Cleaning up, then calling user function.")
+
+        self.timer.stop()
+        result_data = self.api_runner.result
+        self.call_reset()
+
+        user_callback(r, result_data)
+
+
+    def timeout_cleanup(self, user_callback):
+        logger.info("API call timed out. Doing a cleanup, then calling user function.")
+
+        if self.api_thread:
+            self.api_runner.i_timed_out = True
+            self.call_reset()
+
+        user_callback()
+
```

Comment:
Here are our callback wrappers, which reset the state of the Client before calling user callbacks.